### PR TITLE
Merger of 3 different development branches into one PR

### PR DESCRIPTION
**Pull request by alspellm**
Add track states, bfieldy at track state locations, KF isolation values, ZBi Processor for SIMPs

### CODE REVIEW

File: event/include/TrackerHit.h [hunk 1]
```diff
@@ -180,6 +186,9 @@ class TrackerHit : public TObject {
         /** LCIO ids of MC Particles associated to the hit */
         std::vector<int> mcPartIDs_;
 
+        /** RawHit Strip numbers on hit */
+        std::vector<int> rawhits_;
```
Comment (cbravo135) [inline event/include/TrackerHit.h]: Don't these already exist in raw_hits_ of line 166?

File: analysis/selections/simps/Tight_2016_simp_reach_CR.json
Comment (alspellm) [inline analysis/selections/simps/Tight_2016_simp_reach_CR.json:1]: moved chi2 cut to preselection

File: analysis/selections/simps/Tight_2016_simp_reach_SR.json
Comment (alspellm) [inline analysis/selections/simps/Tight_2016_simp_reach_SR.json:1]: moved chi2 cut to preselection

File: analysis/selections/simps/radMatchTight_2016_simp_reach_CR.json
Comment (alspellm) [inline analysis/selections/simps/radMatchTight_2016_simp_reach_CR.json:1]: move chi2 to preselection

File: analysis/src/MCAnaHistos.cxx
Comment (alspellm) [inline analysis/src/MCAnaHistos.cxx:1]: simp PDGID is 625 not 622

File: analysis/src/TrackHistos.cxx
Comment (alspellm) [inline analysis/src/TrackHistos.cxx:1]: if track state at target exists, track position is set. Same for ecal. This adds those positions to histograms (if they exist)
Also added track errors

File: processors/config/anaTrackTuple_cfg.py
Comment (alspellm) [inline processors/config/anaTrackTuple_cfg.py:1]: restored to original state

File: processors/include/VertexAnaProcessor.h
Comment (alspellm) [inline processors/include/VertexAnaProcessor.h:1]: include some corrections, and add run number for saving to flat tuple

File: processors/src/FinalStateParticleProcessor.cxx
Comment (alspellm) [inline processors/src/FinalStateParticleProcessor.cxx:1]: buildParticle takes trackstate location now. Hardcode to primary hps-java track for now (AtIP)

File: processors/src/MCParticleProcessor.cxx
Comment (alspellm) [inline processors/src/MCParticleProcessor.cxx:1]: I think the new hps-java lcio output requires this change. This works for me

File: processors/src/RefittedTracksProcessor.cxx
Comment (alspellm) [inline processors/src/RefittedTracksProcessor.cxx:1]: just adding new track state and defaulting at AtIP

File: processors/src/Tracker2DHitProcessor.cxx
Comment (alspellm) [inline processors/src/Tracker2DHitProcessor.cxx:1]: add rawhit info to hits

File: processors/src/TrackingProcessor.cxx
Comment (alspellm) [inline processors/src/TrackingProcessor.cxx:1]: buildtrack now uses track state location (default is always AtIP)

**Review by cbravo135 — APPROVED** [review]

**Pull request merged**